stdlib/src/locale.rs

```diff
@@ -67,6 +68,33 @@ mod _locale {
         )
     }
 
+    #[pyfunction]
+    fn strcoll(string1: PyStrRef, string2: PyStrRef, vm: &VirtualMachine) -> PyResult {
+        unsafe {
+            let string1_cstr = CString::new(string1.as_str()).map_err(|e| e.to_pyexception(vm))?;
+            let string2_cstr = CString::new(string2.as_str()).map_err(|e| e.to_pyexception(vm))?;
+            Ok(vm.new_pyobj(libc::strcoll(string1_cstr.as_ptr(), string2_cstr.as_ptr())))
+        }
+    }
+
+    #[pyfunction]
+    fn strxfrm(string: PyStrRef, vm: &VirtualMachine) -> PyResult {
+        // https://github.com/python/cpython/blob/eaae563b6878aa050b4ad406b67728b6b066220e/Modules/_localemodule.c#L390-L442
+        unsafe {
+            let n1 = string.byte_len() + 1;
+            let mut buff = vec![0; n1];
+            let buff_ptr = buff.as_mut_ptr();
+
+            let string_cstr = CString::new(string.as_str()).map_err(|e| e.to_pyexception(vm))?;
+            let string_ptr = string_cstr.as_ptr();
+            let n2 = libc::strxfrm(buff_ptr, string_ptr, n1) + 1;
```

Comment:
```suggestion
            let n2 = libc::strxfrm(buff.as_mut_ptr() as _, string_ptr, n1) + 1;
```
Then this call can be rewritten using raw pointer casting.